libs/components/router/src/lib/modules/href/href.directive.ts

```diff
@@ -11,14 +11,37 @@ import {
   Renderer2,
 } from '@angular/core';
 import { Router, UrlTree } from '@angular/router';
-import { SkyAppConfig, SkyAppRuntimeConfigParamsProvider } from '@skyux/config';
+import {
+  SkyAppConfig,
+  SkyAppRuntimeConfigParamsProvider,
+  SkyConfigQueryParams,
+} from '@skyux/config';
 
 import { SkyHrefResolverService } from './href-resolver.service';
 import { SkyHref } from './types/href';
 import { SkyHrefChange } from './types/href-change';
+import { SkyHrefQueryParams } from './types/href-query-params';
 
 type HrefChanges = { href: string; hidden: boolean };
 
+/**
+ * Transforms SkyHrefQueryParams into SkyConfigQueryParams.
+ */
+function toConfigQueryParams(params: SkyHrefQueryParams): SkyConfigQueryParams {
+  // Clone the object to avoid modifying the original.
+  const clone = { ...params };
+
+  for (const key in clone) {
```

Comment:
I would recommend against filtering the query params here because it will create forked behavior based on who calls getLinkUrl.

Reply:
Good point. Reverting.